pkg/operator/etcdcertsigner/etcdcertsignercontroller.go

```diff
@@ -171,16 +184,44 @@ func NewEtcdCertSignerController(
 }
 
 func (c *EtcdCertSignerController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
-	safe, err := c.quorumChecker.IsSafeToUpdateRevision()
+	bootstrapComplete, err := bootstrap.IsBootstrapComplete(c.configmapLister)
 	if err != nil {
-		return fmt.Errorf("EtcdCertSignerController can't evaluate whether quorum is safe: %w", err)
+		return err
 	}
 
-	if !safe {
-		return fmt.Errorf("skipping EtcdCertSignerController reconciliation due to insufficient quorum")
+	hasDiff, err := c.hasNodeCertDiff()
+	if err != nil {
+		return err
+	}
+
+	// we allow the controller to run freely during bootstrap to avoid having issues with constantly rolling out revisions and other
+	// contention issues on the operator status update. Same during vertical scaling, when a node was added we want to immediately issue a cert.
+	if !bootstrapComplete || hasDiff {
+		klog.Infof("EtcdCertSignerController force sync bootstrap completed=[%v], found node difference=[%v]", bootstrapComplete, hasDiff)
+		if err := c.syncAllMasterCertificates(ctx, syncCtx.Recorder(), true, 0, 0); err != nil {
+			return fmt.Errorf("EtcdCertSignerController failed to sync all master certificates during bootstrap: %w", err)
+		}
+		return nil
+	}
+
+	_, currentStatus, _, err := c.operatorClient.GetStaticPodOperatorStateWithQuorum(ctx)
+	if err != nil || currentStatus == nil {
+		return fmt.Errorf("skipping EtcdCertSignerController can't get current status: %w", err)
+	}
+
+	currentRevision, err := ceohelpers.CurrentRevision(*currentStatus)
```

Comment:
we might be able to relax this a bit and return the MIN(all current revisions) and keep the remainder of the controller running. That could save us the `hasNodeCertDiff` call above, but we can also improve this later if we need to.

Reply:
Ah okay, so we would no longer block blindly if there is a revision rollout. That let's us keep the controller going during vertical scaling. But we get the MIN revision that's currently lagging and only compare that against the `openshift.io/ceo-bundle-rollout-revision` value only gate the leaf certs if the MIN revision hasn't moved past the last rotation revision.

Actually now that I wrote that out, that means the controller would still block from generating leaf certs for the new node that gets added. So maybe I don't quite follow how that would help us replace `hasNodeCertDiff()` 😓

Reply:
In any case, if you want to target this as later improvement, I'm good with the `hasNodeCertDiff()` for now. Or if you make the change, I can try and see how that works too.

Reply:
let's test this in a separate PR, otherwise we won't get anything done in here :mechanic:  
I'll have to still debug Vadim's cert rotation workflow, so we can certainly expect one or two more fixes coming out of that :)